fastlane_bot/events/exchanges/solidly_v2.py

```diff
@@ -41,31 +60,23 @@ def add_pool(self, pool: Pool):
         self.pools[pool.state["address"]] = pool
 
     def get_abi(self):
-        return VELOCIMETER_V2_POOL_ABI if "velocimeter" in self.exchange_name else SOLIDLY_V2_POOL_ABI
+        return EXCHANGE_INFO.get(self.exchange_name).get("pool_abi")
 
     @property
     def get_factory_abi(self):
-        return VELOCIMETER_V2_FACTORY_ABI if "velocimeter" in self.exchange_name else SOLIDLY_V2_FACTORY_ABI
+        return EXCHANGE_INFO.get(self.exchange_name).get("factory_abi")
 
     def get_events(self, contract: Contract) -> List[Type[Contract]]:
         return [contract.events.Sync] if self.exchange_initialized else []
 
     async def get_fee(self, address: str, contract: Contract) -> Tuple[str, float]:
-        if "velocimeter" in self.exchange_name:
-            default_fee = await self.factory_contract.caller.getFee(address)
-            default_fee = float(default_fee) / 10000
-
-        elif self.exchange_name in ["scale_v2", "equalizer_v2"]:
-            default_fee = await self.factory_contract.caller.getRealFee(address)
-            default_fee = float(default_fee) / 10 ** 18 if self.exchange_name == "scale_v2" else float(default_fee) / 10000
-        else:
-            is_stable = await contract.caller.stable()
-            default_fee = await self.factory_contract.caller.getFee(address, is_stable)
-            default_fee = float(default_fee) / 10000
-        return str(default_fee), default_fee
+        fee = await EXCHANGE_INFO.get(self.exchange_name).get("fee_function")(address, contract, self.factory_contract)
```

Comment:
You can use `[]` instead of `get`, it is cleaner and more readable.
Also, instead of reading `EXCHANGE_INFO[self.exchange_name]` twice, you can do something like:
```
exchange_info = EXCHANGE_INFO[self.exchange_name]
fee = await exchange_info["fee_function"](address, contract, self.factory_contract)
fee_float = float(fee) / 10 ** exchange_info["decimals"]
```

Reply:
updated